pkg/atls/attestation_provider.go

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) Ultraviolet
+// SPDX-License-Identifier: Apache-2.0
+package atls
+
+import (
+	"encoding/asn1"
+	"fmt"
+
+	"github.com/ultravioletrs/cocos/pkg/attestation"
+	"github.com/ultravioletrs/cocos/pkg/attestation/azure"
+	"github.com/ultravioletrs/cocos/pkg/attestation/tdx"
+	"github.com/ultravioletrs/cocos/pkg/attestation/vtpm"
+	"golang.org/x/crypto/sha3"
+)
+
+// AttestationProvider defines the interface for platform attestation operations.
+type AttestationProvider interface {
+	GetAttestation(pubKey []byte, nonce []byte) ([]byte, error)
```

Comment:
Let's use simply `Attest(pubKey, nonce []byte) ([]byte, error)` signature. Get is usually "getter" or "local", while this fetch is less specific. Do the same for others, no need for `Get` prefix.